subprojects/core/src/main/java/org/gradle/plugin/use/internal/PluginRequestCollector.java

```diff
@@ -109,11 +110,10 @@ public PluginId transform(PluginRequestInternal pluginRequest) {
                 PluginRequestInternal first = pluginRequests.get(0);
                 PluginRequestInternal second = pluginRequests.get(1);
 
-                InvalidPluginRequestException exception = new InvalidPluginRequestException(second, "Plugin with id '" + key + "' was already requested at line " + first.getLineNumber());
-                throw new LocationAwareException(exception, second.getScriptDisplayName(), second.getLineNumber());
+                InvalidPluginRequestException exception = new InvalidPluginRequestException(second, "Plugin with id '" + key + "' was already requested (" + first.getOrigin() + ")");
```

Comment:
⭕️ Do we even use the "origin" anywhere else? If not, let's just remove this check. Several people agreed on different occasions that it serves no real purpose.

Reply:
It's also being used in `DefaultPluginRequestApplicator` in three places which was the main reason why I change the `LocationAwareException` as well.

- https://github.com/gradle/gradle/blob/bm/plugin-request-origin/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java#L242
- https://github.com/gradle/gradle/blob/bm/plugin-request-origin/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java#L267
- https://github.com/gradle/gradle/blob/bm/plugin-request-origin/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java#L273

We could potentially throw a different exception here.

Reply:
I missed those, thanks for the pointers!